src/components/Form.js

```diff
@@ -17,32 +16,46 @@ export default class Form extends Component {
     noValidate: false,
     liveValidate: false,
     safeRenderCompletion: false,
-  }
+  };
 
   constructor(props) {
     super(props);
     this.state = this.getStateFromProps(props);
   }
 
+  componentDidMount() {
+    const {formData} = this.state;
+    const {liveValidate, noValidate, schema} = this.props;
+    const {errors = [], errorSchema = {}} = this.state;
+    const mustValidate = !!formData && !noValidate && liveValidate;
+
+    Promise
```

Comment:
Note that you can totally land the patch using the regular Promise API and migrate the code to async/await in a next PR if that worries you